### DISCUSSION TITLE
A lot off timeout exception

### DISCUSSION THREAD

**Discussion by Alt0car** [tectalichq/public-openai-client-php#33] [Q&A] [answered]
Hello,

As you know, GPT4 takes a very long time to give responses and I very often come up against error timeouts.
Is there any way of passing options to increase this?

`PHP Fatal error:  Uncaught Symfony\Component\HttpClient\Exception\TimeoutException: Idle timeout reached for "https://api.openai.com/v1/chat/completions".`

I've tried this way but without success :

`$sHttp      = HttpClient::create(['timeout' => 600,
                                  'max_duration' => 600
                                 ]);
$httpClient = new Psr18Client($sHttp);
$client     = new Client($httpClient, $auth, Manager::BASE_URI);`

**Comment by thejamescollins** [comment, marked as answer]
Hi there @Alt0car, thanks for getting in touch.

The Tectalic API client itself doesn't have any limits on API request timeouts. Instead, that is the responsibility of the HTTP client that you are using.

By the looks of things you are using the Symfony PSR18 HTTP client.

Looking at https://symfony.com/doc/current/http_client.html#psr-18-and-psr-17, it seems you may need to use code similiar to this:

```php
$httpClient = (new Psr18Client())->withOptions([  'timeout' => 600, 'max_duration' => 600 ]);

$client     = new Client($httpClient, $auth, Manager::BASE_URI);
```

Good luck with your project!

James
